### ISSUE TITLE
storage upgrade from 1.2.1 to LEGACY branch doesn't upgrade correctly

### ISSUE THREAD

**Issue by frankenthumbs** [react-native-community/async-storage#301]
## Current behavior

when upgrading storage from 1.2.1 to the LEGACY branch, storage from an old directory is not being moved/copied over to the new directory... i believe this is happening because on `init` when `RCTStorageDirectoryMigrate` tries to migrate new data to `[root]/data/Containers/Data/Application/[app id]/Library/Application Support/[bundle id]/RCTAsyncLocalStorage_V1` the folder doesn't exist. when i try to run the below code right before the filemanager attempts to copy over the data, migration `[NSFileManager copyItemAtPath: toPath: Error:]` runs correctly:
```
    if (!RCTHasCreatedStorageDirectory) {
        NSString *storageDirectoryPath = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES).firstObject;
        storageDirectoryPath = [storageDirectoryPath stringByAppendingPathComponent:[[NSBundle mainBundle] bundleIdentifier]];

      [[NSFileManager defaultManager] createDirectoryAtPath:storageDirectoryPath
                                withIntermediateDirectories:YES
                                                 attributes:nil
                                                      error:&error];
      RCTHasCreatedStorageDirectory = YES;
    }
```

![Screen Shot 2020-02-18 at 6 34 07 PM](https://user-images.githubusercontent.com/3641335/74796589-e5eb1580-527d-11ea-97d4-93d36cb56945.png)

## Expected behavior

We shouldn't get the warning and folder migration should happen correctly.

## Repro steps

1. check out this repo: https://github.com/frankenthumbs/AsyncStorageUpgradePOC
2. run `yarn` on root
3. run `pod install` on `/ios` folder
4. run ios application
5. tap `set storage to 'I Like to Save It'` to save a string to async storage
6. in package json, upgrade `@react-native-community/async-storage` to `git://github.com/react-native-community/async-storage.git#4e49db6308882515a284a183c8e9daf5a50b33ef`
7. in change lines
```
pod 'react-native-async-storage', :path => '../node_modules/@react-native-community/async-storage'
```
to 
```
pod 'RNCAsyncStorage', :path => '../node_modules/@react-native-community/async-storage'
```
8. run `yarn` and `pod install` described in step 2 and 3
9. run application and notice that async-storage is null (where it should be set to saved string) and we get aforementioned warning.

## Environment

- **Async Storage version**:  commit: 4e49db6308882515a284a183c8e9daf5a50b33ef
- **React-Native version**: 0.61.5
- **Platform tested**: iOS
- **Logs/Error that are relevant**: ```2020-02-18 17:00:13.828 [warn][tid:com.facebook.react.JavaScript][RNCAsyncStorage.m:205] Failed to copy old storage directory to new storage directory location during migration: Error Domain=NSCocoaErrorDomain Code=4 "The file “RCTAsyncLocalStorage_V1” doesn’t exist." UserInfo={NSSourceFilePathErrorKey=/Users/joon.son/Library/Developer/CoreSimulator/Devices/A9A28D7F-CDD1-49EC-9138-356C0A9A7C4C/data/Containers/Data/Application/B8CAFA4D-89A5-4EE5-92F0-78B4406870D3/Documents/RCTAsyncLocalStorage_V1, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/joon.son/Library/Developer/CoreSimulator/Devices/A9A28D7F-CDD1-49EC-9138-356C0A9A7C4C/data/Containers/Data/Application/B8CAFA4D-89A5-4EE5-92F0-78B4406870D3/Library/Application Support/com.nfl.mobile3/RCTAsyncLocalStorage_V1, NSFilePath=/Users/joon.son/Library/Developer/CoreSimulator/Devices/A9A28D7F-CDD1-49EC-9138-356C0A9A7C4C/data/Containers/Data/Application/B8CAFA4D-89A5-4EE5-92F0-78B4406870D3/Documents/RCTAsyncLocalStorage_V1, NSUnderlyingError=0x60000159a3a0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}```

**Comment by lorenc-tomasz** [comment]
Have the same issue.

**Comment by frankenthumbs** [comment]
i put up a PR here. https://github.com/react-native-community/async-storage/pull/303 definitely not my finest work.. would love feedback on it if possible.

**Comment by christianrank** [comment]
Also getting this issue after switching from React Native's internal Async Storage implementation to this external package. 🙁

**Comment by christianrank** [comment]
Still getting the issue on 1.7.1, because the Pod RNCAsyncStorage 1.8.0 is installed with it. How can I switch back to the 1.7.1 pod?

Doing it manually in podfile.lock doesn't work.

```
[!] CocoaPods could not find compatible versions for pod "RNCAsyncStorage":
  In snapshot (Podfile.lock):
    RNCAsyncStorage (from `../node_modules/@react-native-community/async-storage`)

  In Podfile:
    RNCAsyncStorage (from `../node_modules/@react-native-community/async-storage`)


You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * changed the constraints of dependency `RNCAsyncStorage` inside your development pod `RNCAsyncStorage`.
   You should run `pod update RNCAsyncStorage` to apply changes you've made.
```

**Comment by tranhoangduong1994** [comment]
I experienced the same issue with 1.8.0. For now, using 1.7.1 is my workaround.

**Comment by frankenthumbs** [comment]
should be fixed with 1.8.1. closing.
